File: hash.js
var i = 0;
var h = {};

//for(i = 0; i < 10000000; i++) {
while(i < 10000000) {
  h[i] = "aaa" + i;
  //if(i < 100) {
    //console.log(sprintf("aaa%d", i));
  //}
  i += 1;
}

// 6s, ~580MB RAM

File: hash.py
i = 0
h = {}
while i < 1000000:
  h[i] = "aaa" + str(i)
  if i < 100:
    #print("aaa{0}".format(i))
    print(h[i])
  i += 1

# 1.35s, ~130MB RAM